mail/src/main/java/org/jboss/as/mail/extension/MailSessionAdd.java

```diff
@@ -93,16 +93,17 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
     private static void addCredentialStoreReference(ServerConfig serverConfig, OperationContext context, ModelNode model, ServiceBuilder serviceBuilder, String... modelFilter) throws OperationFailedException {
```

Comment:
Unless @yersan disagrees, I think this is a bad method signature. The method requires 2 non-null string params, not an arbitrary length array. All the callers (correctly) pass two strings because that's the model structure this method is meant to process.

Also the 'break' at L102 should be a 'return'. If the model doesn't have the expected structure processing should stop.

Reply:
@bstansberry The tests with two Strings and && logic fails now. It seems [MailSessionAdd#L132](https://github.com/wildfly/wildfly/blob/2bfac3db7e000ae152872c2a6ba08c3ae4412eaa/mail/src/main/java/org/jboss/as/mail/extension/MailSessionAdd.java#L132) leads to actual "fallback logic" with the break.